oh-my-zsh.sh

```diff
@@ -143,37 +142,45 @@ fi
 
 unset zcompdump_revision zcompdump_fpath zcompdump_refresh
 
-
 # Load all of the config files in ~/oh-my-zsh that end in .zsh
 # TIP: Add files you don't want in git to .gitignore
-for config_file ($ZSH/lib/*.zsh); do
-  custom_config_file="${ZSH_CUSTOM}/lib/${config_file:t}"
-  [ -f "${custom_config_file}" ] && config_file=${custom_config_file}
-  source $config_file
+for config_file ("$ZSH"/lib/*.zsh); do
+  custom_config_file="$ZSH_CUSTOM/lib/${config_file:t}"
+  [[ -f "$custom_config_file" ]] && config_file="$custom_config_file"
+  source "$config_file"
 done
 
 # Load all of the plugins that were defined in ~/.zshrc
 for plugin ($plugins); do
-  if [ -f $ZSH_CUSTOM/plugins/$plugin/$plugin.plugin.zsh ]; then
-    source $ZSH_CUSTOM/plugins/$plugin/$plugin.plugin.zsh
-  elif [ -f $ZSH/plugins/$plugin/$plugin.plugin.zsh ]; then
-    source $ZSH/plugins/$plugin/$plugin.plugin.zsh
+  if is_plugin "$ZSH_CUSTOM" "$plugin"; then
+    source "$ZSH_CUSTOM/plugins/$plugin/$plugin.plugin.zsh"
+  elif is_plugin "$ZSH" "$plugin"; then
+    source "$ZSH/plugins/$plugin/$plugin.plugin.zsh"
```

Comment:
Now that I look more carefully, this is not totally correct because `is_plugin` will return correctly if a `_$plugin` file exists but not the `$plugin.plugin.zsh`, so this will break on plugins that only provide completion files.

It should only be used in the `fpath` change logic above.

Reply:
uff I didn't caught that. I'll revise it!

Reply:
I think that what you say is not correct Marc.
```zsh
is_plugin() {
  local base_dir=$1
  local name=$2
  builtin test -f $base_dir/plugins/$name/$name.plugin.zsh \
    || builtin test -f $base_dir/plugins/$name/_$name
}
```
This will return correctly in both cases. Maybe we should modify the logic here

Reply:
That would change how we add plugins to `$fpath`.

If we only return true if `$name.plugin.zsh` exists, plugins without these files (like most completion plugins) will not get added to `$fpath` and break.
If we only return true if `_$name` exists, plugins without `_$name` like the `genpass` plugin will not work correctly if they're not added to `$fpath`.

Also the proposed change does not decouple the path to be sourced from the check if the file exists, so it moves the logic to two places (`is_plugin` and the `source` line).

We could turn this into a `for` loop but given that we only need to check 2 locations that seems wasteful. It's better to leave it as is.

Reply:
Yeah that's the problem. We could also have a function check if the plugin file exists and loads it like you say, but we'd only be moving the logic somewhere else.